test/async-hooks/test-zctx.js

```diff
@@ -0,0 +1,31 @@
+const common = require('../common')
+const assert = require('assert')
+const initHooks = require('./init-hooks')
+const { checkInvocations } = require('./hook-checks')
+const zlib = require('zlib')
+
+const hooks = initHooks()
+
+hooks.enable()
+zlib.deflate('..........', common.mustCall(ondeflate));
+
```

Comment:
Please also check if you have any ztcx resources created synchronously.
If so please collect all info here and assign the resource to a variable that you can reuse to check for invocations.
Do the general handle checks you currently do inside `process.exit` here as well in that case.